content/rooms/fragments/v2-state-res.md

```diff
@@ -82,27 +82,41 @@ selecting, among all the candidate vertices, the smallest vertex using
 the above comparison relation.
 
 **Mainline ordering.**
-Given an `m.room.power_levels` event *P*, the *mainline of* *P* is the
-list of events generated by starting with *P* and recursively taking the
-`m.room.power_levels` events from the `auth_events`, ordered such that
-*P* is last. Given another event *e*, the *closest mainline event to*
-*e* is the first event encountered in the mainline when iteratively
-descending through the `m.room.power_levels` events in the `auth_events`
-starting at *e*. If no mainline event is encountered when iteratively
-descending through the `m.room.power_levels` events, then the closest
-mainline event to *e* can be considered to be a dummy event that is
-before any other event in the mainline of *P* for the purposes of
-condition 1 below.
+Let *P* = *P*<sub>0</sub> be an `m.room.power_levels` event.
+Starting with *i* = 0, repeatedly fetch *P*<sub>*i*+1</sub>, the
+`m.room.power_levels` event in the `auth_events` of *P<sub>i</sub>*.
+Increment *i* and repeat until *P<sub>i</sub>* has no `m.room.power_levels`
+event in its `auth_events`.
+The *mainline of P*<sub>0</sub> is the list of events
+    [*P*<sub>n</sub> , ... , *P*<sub>1</sub>, *P*<sub>0</sub>],
+ordered from oldest to newest.
+
+Let *e* = *e<sub>0</sub>* be another event (possibly another
+`m.room.power_levels` event). We can compute a similar list of events
+    [*e*<sub>1</sub>, ..., *e<sub>m</sub>*],
+where *e<sub>*j* + 1</sub>* is the `m.room.power_levels` event in the
+`auth_events` of *e<sub>j</sub>* and where *e<sub>m</sub>* has no
+`m.room.power_levels` event in its `auth_events`. (Note that this second list
+may be empty, because *e* may not cite an `m.room.power_levels` event in its
+`auth_events` at all.)
+
+Now compare these two lists as follows.
+* Find the smallest index *j* ≥ 1 for which *e<sub>j</sub>* belongs to the
+   mainline of *P*.
+* If *j* exists, then *e<sub>j</sub>* = *P<sub>i</sub>* for some unique index
+  *i* ≥ 0. Otherwise set *i* = ∞, where ∞ is a sentinel value greater
+  than any integer.
```

Comment:
Nit.

```suggestion
* If such a *j* exists, then *e<sub>j</sub>* = *P<sub>i</sub>* for some unique
  index *i* ≥ 0. Otherwise set *i* = ∞, where ∞ is a sentinel value greater
  than any integer.
```